prow/github/types.go

```diff
@@ -357,6 +357,16 @@ type Branch struct {
 	// TODO(fejta): consider including undocumented protection key
 }
 
+// BranchProtection represents protections
+// in place for a branch
+// See also: https://developer.github.com/v3/repos/branches/#get-branch-protection
+type BranchProtection struct {
+	RequiredStatusChecks       *RequiredStatusChecks       `json:"required_status_checks"`
+	EnforceAdmins              *bool                       `json:"enforce_admins"`
+	RequiredPullRequestReviews *RequiredPullRequestReviews `json:"required_pull_request_reviews"`
+	Restrictions               *Restrictions               `json:"restrictions"`
+}
```

Comment:
This struct doesn't appear to match the [payload example](https://developer.github.com/v3/repos/branches/#response-2). In particular the `EnforceAdmins` field.

Are these fields always present in the payload? If so we should use struct types instead of pointers to structs.

Reply:
Thank you for the review. From my tests it seems all fields, except `EnforceAdmins`, can be absent. I will update the PR to address this.

Reply:
I see, fixed it. `enforce_admins` is always present in the payload in my tests, so will use a struct. All other fields can be missing from the payload.

Reply:
nit: would want `omitempty` on optional fields for serialization